core/trino-main/src/main/java/io/trino/operator/scalar/MathFunctions.java

```diff
@@ -1176,6 +1176,20 @@ public static boolean isNaN(@SqlType(StandardTypes.DOUBLE) double num)
         return Double.isNaN(num);
     }
 
+    @Description("Test if value is not-a-number")
+    @ScalarFunction("is_nan")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean isNaNReal(@SqlType(StandardTypes.REAL) long value)
+    {
+        try {
+            float floatValue = intBitsToFloat(toIntExact(value));
+            return Float.isNaN(floatValue);
+        }
+        catch (IllegalArgumentException e) {
+            throw new TrinoException(NUMERIC_VALUE_OUT_OF_RANGE, "Out of range for integer: " + value, e);
```

Comment:
No need to produce TrinoException for what would be an internal bug (or a bug in a connector)
(just like some ArrayIndexOutOfBound, or NPE).

Reply:
Removed